org.civicrm.reqangular/src/common/services/format-currency.service.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/services',
+  'common/services/api',
+  'common/models/settings.model'
+], function (services) {
+  'use strict';
+
+  services.factory('FormatCurrencyService', FormatCurrencyService);
+
+  FormatCurrencyService.$inject = ['$log', '$q', 'Settings'];
+
+  function FormatCurrencyService ($log, $q, Settings) {
+    $log.debug('Service: FormatCurrencyService');
+
+    return {
+      format: format
+    };
+
+    /**
+    * Adds defined thousand and decimal separators in the given amount
+    *
+    * @param {string} amount
+    * @return {promise}
+    */
+    function format (amount) {
+      return Settings.fetchSeparators().then(function (separators) {
+        return {
+          formatted: addSeparators(amount, separators),
+          unformatted: removeSeparators(amount, separators)
+        };
+      });
+    }
+
+    /**
+     * Removes characters other than decimal separator
+     * Converts the formatted amount to numeric value (making calulable) by
+     * removing the thousand separators and replacing the decimal separator by " ."
+     *
+     * @param  {string|Integer} amount
+     * @param  {object} separators
+     * @return {integer}
+     */
+    function removeSeparators (amount, separators) {
+      var expression = configureExpression(separators.decimal);
+      var strippedAmount = amount.toString().replace(new RegExp(expression, 'g'), '');
+      var amountWithDecimal = strippedAmount.replace(separators.decimal, '.');
+
+      return amountWithDecimal;
+    }
+
+    /**
+     * Configure expression for special characters like \ and ]
+     *
+     * @param  {string} decimalSeparator
+     * @return {string}
+     */
+    function configureExpression (decimalSeparator) {
```

Comment:
this function has a very vague name. What expression? for what? the name doesn't say

Reply:
Agreed! 
We should better call it `prepareRegexExpression` or `expressionWithEscapedCharacters` what do you suggest?

cc @igorpavlov

Reply:
`prepareRegexForSeparator` ? cc @bmnepali @AkA84

Reply:
`prepareRegexForSeparator ` looks better.